lib/backends/sdc/build.js

```diff
@@ -891,8 +894,266 @@ function runBuildCommand(opts, callback) {
+                        function _imageCreateCb(err, result)
+                    {
+                        sendEventResponse(event, err, result);
+                    });
+                    break;
+                case 'commit_finished':
+                    finalImageId = event.finalId;
+                    vasync.pipeline({ funcs: [
+                        function _doAddBaseImageHead(_result, next) {
+                            addBaseImageHead(baseImageId, finalImageId,
+                            {
+                                baseImageIndexName: baseImg.index_name,
+                                req: req,
+                                scratchImage: scratchImage
+                            }, next);
+                        },
+                        function _doTagImage(_result, next) {
+                            tagImage({
+                                    docker_id: finalImageId,
+                                    name: opts.tag,
+                                    req: req
+                                }, next);
+                        }
+                    ]}, function commitFinishPipeCb(err) {
+                        if (!err) {
+                            commitFinishedSuccessfully = true;
+                        }
+                        sendEventResponse(event, err);
+                    });
+                    break;
+                default:
+                    log.error('Unhandled commit event: %j', event);
+                    break;
+            }
+        });
+        commitEventStream.on('close', function _eventStreamCloseCb() {
+            log.debug('commit: commitEventStream.close - closing socket');
+        });
+        commitSocket.on('end', function _commitSocketEndCb() {
+            log.debug('commit: commitSocket.end', commitError);
+            cb(commitError);
+        });
+
+        log.debug('commit: creating second connection to cnapi');
+    }
+
+
+    function commitCleanup(err, results) {
+        log.debug('commit: final callback, err: %j', err);
+        clearTimeout(cleanupTimeoutId);
+        if (err) {
+            callback(err);
+            return;
+        }
+        if (previousErr) {
+            // Error occurred during a previous cn-agent task.
+            callback(previousErr);
+            return;
+        }
+        if (!commitFinishedSuccessfully) {
+            callback(new errors.DockerError('Unexpected commit error'));
+            return;
+        }
+        callback(null, finalImageId);
+    }
+
+
+    function recordError(err, logMessage) {
+        log.error(err, logMessage);
+        if (!previousErr) {
+            previousErr = err;
+            // If in 5 minutes the cn-agent task hasn't gotten back to us, then
+            // something is off the rails and we'll abort/cleanup from here by
+            // firing the callback, thus closing the client.
+            cleanupTimeoutId = setTimeout(commitCleanup, 5 * 60 * 1000);
```

Comment:
We might be able to not have to setup a custom timer by adding a `'timeout'` event listener on the socket as mentioned above. If the default timeout set by node for sockets is not suitable, we can set a custom timeout value with `setTimeout(timeoutValue)` on the socket.